nmrfx-utils/src/test/java/org/nmrfx/utils/properties/ListOperationItemTest.java

```diff
@@ -0,0 +1,171 @@
+        Assert.assertEquals(lstItem.value, lstItem.defaultValue);
+        Assert.assertEquals(lstItem.value, List.of(1.0, 2.0, 3.0, 4.0));
+    }
+
+    @Test
+    public void testGetValueArrayListNumber() {
+        ArrayList<Double> defaultList = new ArrayList<>(List.of(1.0, 2.0, 3.0, 4.0));
+        ListOperationItem lstItem = new ListOperationItem(listListener, defaultList, "COADD", "coef", "List of coefficients to scale each vector by.", typeSelector);
+        String value = lstItem.getValue();
+        Assert.assertEquals("[1.0, 2.0, 3.0, 4.0]", value);
+    }
+
+    @Test
+    public void testGetValueVectorNumber() {
+        Vector<Double> defaultList = new Vector<>(List.of(1.0, 2.0, 3.0, 4.0));
+        ListOperationItem lstItem = new ListOperationItem(listListener, defaultList, "COADD", "coef", "List of coefficients to scale each vector by.", typeSelector);
+        String value = lstItem.getValue();
+        Assert.assertEquals("[1.0, 2.0, 3.0, 4.0]", value);
+    }
+
+    @Test
+    public void testSetValueString() {
+        ArrayList<Double> defaultList = new ArrayList<>(List.of(1.0, 2.0, 3.0, 4.0));
+        ListOperationItem lstItem = new ListOperationItem(listListener, defaultList, "COADD", "coef", "List of coefficients to scale each vector by.", typeSelector);
+        String newValue = "[5.0, 6.0, 7.0, 8.0]";
+        lstItem.setValue(newValue);
+        Assert.assertEquals(List.of(5.0, 6.0, 7.0, 8.0), lstItem.value);
+        Assert.assertNotEquals(lstItem.value, lstItem.defaultValue);
+        Assert.assertEquals(1, changeListenerCallCount);
+    }
+
+    @Test
+    public void testSetValueArrayList() {
+        ArrayList<Double> defaultList = new ArrayList<>(List.of(1.0, 2.0, 3.0, 4.0));
+        ListOperationItem lstItem = new ListOperationItem(listListener, defaultList, "COADD", "coef", "List of coefficients to scale each vector by.", typeSelector);
+        ArrayList<Double> newValue = new ArrayList<>(List.of(5.0, 6.0, 7.0, 8.0));
+        lstItem.setValue(newValue);
+        Assert.assertEquals(List.of(5.0, 6.0, 7.0, 8.0), lstItem.value);
+        Assert.assertNotEquals(lstItem.value, lstItem.defaultValue);
+        Assert.assertEquals(1, changeListenerCallCount);
+    }
+
+    @Test
+    public void testSetValueArrayListSameValues() {
+        ArrayList<Double> defaultList = new ArrayList<>(List.of(1.0, 2.0, 3.0, 4.0));
+        ListOperationItem lstItem = new ListOperationItem(listListener, defaultList, "COADD", "coef", "List of coefficients to scale each vector by.", typeSelector);
+        ArrayList<Double> newValue = new ArrayList<>(List.of(1.0, 2.0, 3.0, 4.0));
+        lstItem.setValue(newValue);
+        Assert.assertEquals(List.of(1.0, 2.0, 3.0, 4.0), lstItem.value);
+        Assert.assertEquals(lstItem.value, lstItem.defaultValue);
+        // Listener is not updated if the values have not changed
+        Assert.assertEquals(0, changeListenerCallCount);
```

Comment:
Good initiative adding unit tests!

Minor things: 
* you could `import static import static org.junit.Assert.*` to avoid prefixing all assertions
* you can replace your comment by a message in the assertion (as the first parameter to asserEquals)

Reply:
Made these two changes